HW17/Vector2d.cpp

```diff
@@ -0,0 +1,108 @@
+#include "Vector2d.h"
+#include <cmath>
+
+int Vector2d::count = 0;
+
+Vector2d::Vector2d()
+{
+	x = 0;
+	y = 0;
+
+	count++;
+}
+
+Vector2d::Vector2d(float x, float y)
+{
+	this->x = x;
+	this->y = y;
+
+	count++;
+}
+
+Vector2d::Vector2d(const Vector2d& other)
+{
+	this->x = other.x;
+	this->y = other.y;
+
+	count++;
+}
+
+Vector2d& Vector2d::operator=(const Vector2d& other)
+{
+	if (this != &other) {
+		this->x = other.x;
+		this->y = other.y;
+	}
+
+	count++;
+
+	return *this;
+}
+
+Vector2d Vector2d::operator+(const Vector2d& secondVector)
+{
+	Vector2d vector = Vector2d(this->x + secondVector.x, this->y + secondVector.y);
+
+	return vector;
+}
+
+Vector2d Vector2d::operator-(const Vector2d& secondVector)
+{
+	Vector2d vector = Vector2d(this->x - secondVector.x, this->y - secondVector.y);
+
+	return vector;
+}
+
+void Vector2d::operator*=(float scalar)
+{
+	x *= scalar;
+	y *= scalar;
+}
+
+Vector2d operator+(const Vector2d& leftVector, const Vector2d& rightVector)
+{
+	Vector2d vector = Vector2d(leftVector.x + rightVector.x, leftVector.y + rightVector.y);
+
+	return vector;
+}
+
+Vector2d operator-(const Vector2d& leftVector, const Vector2d& rightVector)
+{
+	Vector2d vector = Vector2d(leftVector.x - rightVector.x, leftVector.y - rightVector.y);
+
+	return vector;
+}
+
+float Vector2d::operator()()
+{
+	return std::sqrt(std::pow(x, 2) + std::pow(y, 2));
+}
+
+const float& Vector2d::operator[](std::size_t idx) const
+{
+	if (idx == 0)
+	{
+		return x;
+	}
+	else if (idx == 1)
+	{
+		return y;
+	}
```

Comment:
`not all control paths return value` that what compiler should've warned you about )